salt/cloud/clouds/ec2.py

```diff
@@ -1199,15 +1199,16 @@ def _get_subnetname_id(subnetname):
     for subnet in aws.query(params, location=get_location(),
                provider=get_provider(), opts=__opts__, sigver='4'):
         tags = subnet.get('tagSet', {}).get('item', {})
-        if not isinstance(tags, list):
-            tags = [tags]
-        for tag in tags:
-            if tag['key'] == 'Name' and tag['value'] == subnetname:
-                log.debug(
-                    'AWS Subnet ID of %s is %s',
-                    subnetname, subnet['subnetId']
-                )
-                return subnet['subnetId']
+        if 'tagSet' in subnet:
```

Comment:
With this change `tags = subnet.get('tagSet', {}).get('item', {})` should be moved into the `if` block since there is no reason to get it if we have already determined it isn't in there.

Also this code appears to be expecting tags to be a list but subnet.get('tagSet', {}).get('item', {}) returns a empty dict if there is no is no 'tagSet' or 'item'  in the `tagSet`. The second .get should return an empty list instead

```python
subnet.get('tagSet', {}).get('item', [])
```
and since we need always need to get the `tagSet` if it is in the subnet then we don't need first do a look up to check if it is in the dict and then do another look up to get it:

So it can all be changed to:

```python
        tags = subnet.get('tagSet', {}).get('item', [])
        if tags:
            if not isinstance(tags, list):
                tags = [tags]
            for tag in tags:
                if tag['key'] == 'Name' and tag['value'] == subnetname:
                    log.debug(
                        'AWS Subnet ID of %s is %s',
                        subnetname, subnet['subnetId']
                    )
                    return subnet['subnetId']
```